kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -85,6 +86,12 @@ public TransactionBuilder withTransactionId(
         return this;
     }
 
+    @Override
+    public TransactionBuilder withTableProperties(Engine engine, Map<String, String> properties) {
+        this.tableProperties = Optional.of(properties);
```

Comment:
Do we need to make a copy of the properties? Use `Collections.unmodifiableMap` if needed

Reply:
I think we don't need to modify the properties here? We just need to access it. So we don't need to make a copy here?